### PR TITLE
Chaos TC_1 emqx_node_shutdown

### PR DESCRIPTION
**Pull request by qzhuyan**

### CODE REVIEW

File: bin/chaos_test.sh [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    $BASEDIR/inject_fault.sh "$cluster" "emqx-node-shutdown" || die "stopped: fault injection failed"
+    # step 4: wait for traffic to back to normal
+    sleep 300;
+    # step 5: collect logs
+    $BASEDIR/send_cmd.sh "$cluster" "collect_logs" || die "stopped: collect_logs failed"
+
+    echo "test finished"
+}
+
+# cluster name 'CDK_EMQX_CLUSTERNAME' when you deploy cdk
```
Comment (thalesmg) [inline bin/chaos_test.sh:34]: Nit: looks like this comment is about `$1` and not `$2`, which appears to be the test case name?

File: bin/send_cmd.sh [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+            --timeout-seconds 60 --max-concurrency "30" --max-errors "0"
+
+        aws ssm send-command --document-name "$doc_name" \
+            --parameters "{\"Host\": [\"$LB\"], \"Command\":[\"pub\"],\"Prefix\":[\"cdk\"],\"Topic\":[\"topic_a\"],\"Clients\":[\"200000\"],\"Interval\":[\"50\"]}" \
+            --targets "[{\"Key\":\"tag:cluster\",\"Values\":[\"$cluster\"]},{\"Key\":\"tag:service\",\"Values\":[\"loadgen\"]}]" \
+            --timeout-seconds 60 --max-concurrency "30" --max-errors "0"
+        ;;
+    "collect_logs")
+        aws ssm send-command --document-name "$doc_name" \
+            --parameters "{\"Bucket\":[\"emqx-cdk-cluster\"],\"Path\":[\"$cluster\"]}" \
```
Comment (thalesmg) [inline bin/send_cmd.sh:52]: ~Curiosity: should there be a `Command` key in those parameters, like `pub` and `sub` above?~ Nevermind. :slightly_smiling_face:

Comment (thalesmg) [reply bin/send_cmd.sh:52]: If the object path is only the cluster name, subsequent runs will overwrite previous logs, right?  Would it make sense to namespace the experiment run with some unique identifier, like the start datetime or similar? 

Edit: I saw below that this command preserves the file name of the output generated by node_dump.  If it is unique, then nevermind this comment. :slightly_smiling_face:

File: cdk_emqx_cluster/chaos_test.py [hunk 3]
```diff
@@ -0,0 +1,201 @@
...
+                               core.CfnTag(key='service', value=service),
+                               core.CfnTag(key='cmd', value=construct_id)
+                               ],
+                         document_format='YAML',
+                         document_type='Command',
+                         content=content,
+                         **kwargs)
+
+
+class CdkExpirement(fis.CfnExperimentTemplate):
```
Comment (thalesmg) [inline cdk_emqx_cluster/chaos_test.py]: Nit: `CdkExperiment`?

**Review by thalesmg — APPROVED** [review]

File: bin/send_cmd.sh (hunk 2 above)
Comment (qzhuyan) [reply bin/send_cmd.sh:52]: You are right, node_dump  generates a unique log filename. Also if we integrate with CI, we could use the job name as the cluster name.

**Pull request merged**